update-agent/src/main.rs

```diff
@@ -252,6 +278,15 @@ fn run(args: &Args) -> eyre::Result<()> {
         info!("running update for component `{}`", component.name());
         component
             .run_update(target_slot, &claim, settings.recovery)
+            .inspect(|_| {
+                if let Some(conn) = dbus_update_conn {
+                    interfaces::update_component_state(
+                        component.name(),
+                        ComponentState::Installed,
+                        conn,
```

Comment:
try passing around the InterfaceRef instead of the connection itself. It makes it a bit more clear what this function is doing.